json/src/spec/builtin.rs

```diff
@@ -57,6 +57,41 @@ pub struct AltBn128Pairing {
 	pub pair: u64,
 }
 
+
+/// Bls12 pairing price
+#[derive(Debug, PartialEq, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Bls12PairingPrice {
+	/// Price per final exp
+	pub base: u64,
+	/// Price per pair (Miller loop)
+	pub pair: u64,
+}
+
+/// Pricing for constant Bls12 operations (ADD and MUL in G1 and G2, as well as mappings)
+#[derive(Debug, PartialEq, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Bls12ConstOperationsPrice {
+	/// Fixed price.
+	pub price: u64,
+}
+
+/// Pricing for constant Bls12 operations (ADD and MUL in G1, as well as mappings)
+#[derive(Debug, PartialEq, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Bls12MultiexpPriceG1 {
```

Comment:
Perhaps rename to `Bls12G1Multiexp` for consistency with the structs immediately above this one (which don't have `Price` as a suffix), and bringing `G1` before `Multiexp` in line with naming within the `lib.rs` file.